src/compiler/checker.ts

```diff
@@ -15790,6 +15793,26 @@ namespace ts {
             return type;
         }
 
+        function instantiateReverseMappedType(type: ReverseMappedType, mapper: TypeMapper) {
+            const innerMappedType = instantiateType(type.mappedType, mapper);
+            if (!(getObjectFlags(innerMappedType) & ObjectFlags.Mapped)) {
+                return type;

```

Comment:
It feels a _bit_ odd to return the uninstantiated type, but when we can't perform the mapping post-instantiation, it's our best option, and preserves our current behaviors more (since we never instantiated the type prior to this). The most relevant time this occurs is with the `permissiveMapper` - it guarantees all homomorphic mapped types (that are still generic) map to `any`, which is great and all, but prevents us from forming a reverse mapped type (which, in turn, causes us to give paradoxical conditional type `extends` check results). So we kinda just have to skip that instantiation.

Reply:
Do you have an example where this happens, i.e. where instantiation no longer produces a mapped type?

Reply:
I mean, other that with `permissiveMapper`.

Reply:
The `wildcardType` and the `errorType` are the only two types that make a mapped type outright collapse; so the `permissiveMapper` and scenarios involving errors are likely the only cases.

Reply:
Well, technically any type that isn't a union and doesn't meet the mask
```ts
t.flags & (TypeFlags.AnyOrUnknown | TypeFlags.InstantiableNonPrimitive | TypeFlags.Object | TypeFlags.Intersection)
```
too. But that mask covers pretty much everything. I think `never` isn't included in it, so `never` will also collapse the mapped type (to `never`)?